src/stan/mcmc/hmc/nuts/base_nuts.hpp

```diff
@@ -152,9 +151,9 @@ namespace stan {
         values.push_back(this->energy_);
       }
 
-      bool compute_criterion(Eigen::VectorXd& p_sharp_minus,
-                             Eigen::VectorXd& p_sharp_plus,
-                             Eigen::VectorXd& rho) {
+      virtual bool compute_criterion(Eigen::VectorXd& p_sharp_minus,
```

Comment:
This only needs to be virtual if you are (a) using OO and passing references to the base class, OR (b) using templating and don't implement `compute_criterion` in all subclasses.